examples/restaurant_demo/restaurant.py

```diff
@@ -0,0 +1,308 @@
+        visits["Time entered"] = (
+            pd.to_datetime(visits["Time entered"])
+            .apply(lambda x: x.strftime(r"%H%M"))
+            .astype(int)
+        )
+
+        # Only select the visits that happened during the valid hours
+        valid_visits = visits[visits["Time entered"].isin(VALID_HOURS)]
+
+        # Only count the hours without the minutes (12:00 and 12:30 both add 1 to the 12 count)
+        valid_visits["Time entered"] = valid_visits["Time entered"].apply(
+            lambda x: int(str(x)[:2]) if len(str(x)) == 4 else int(str(x)[:1])
+        )
+
+        for time in valid_visits["Time entered"].unique():
+            hours_count[time] = valid_visits[valid_visits["Time entered"] == time][
+                "Time entered"
+            ].count()
+
+        return hours_count
+
+    def get_private_counts_per_hour(self, epsilon: float = None) -> dict:
+        """Compute an anonymized (within a given threshold of epsilon) version of the number of visits per hour. Return a dictionary mapping hours to number of visits"""
+        private_hours_count = dict()
+
+        # Only count the hours without the minutes (12:00 and 12:30 both add 1 to the 12 count)
+        visits = self._hour_visits.copy()
+        visits["Time entered"] = (
+            pd.to_datetime(visits["Time entered"])
+            .apply(lambda x: x.strftime(r"%H"))
+            .astype(int)
+        )
+
+        # Use the default epsilon value if it is not given as an argument
+        if not epsilon:
+            x = dp.CountInt(self._epsilon)
+        else:
+            x = dp.CountInt(epsilon)
+
+        for time in visits["Time entered"].unique():
+            private_hours_count[time] = x.result(
+                list(visits[visits["Time entered"] == time]["Time entered"].astype(int))
+            )
+
+        return private_hours_count
+
+    def get_non_private_counts_per_day(self) -> dict:
+        """Compute the number of visits per day without any differential privacy. Return a dictionary mapping days to number of visits"""
+        day_counts = dict()
+
+        for day in self._day_visits["Day"].unique():
+            day_counts[day] = self._day_visits[self._day_visits["Day"] == day][
+                "Day"
+            ].count()
+
+        return day_counts
+
+    def get_private_counts_per_day(self, epsilon: float = None) -> dict:
+        """Compute an anonymized (within a given threshold of epsilon) version of the number of visits per day. Return a dictionary mapping days to number of visits"""
+        # Pre-process the data set: limit the number of days contributed by a visitor to COUNT_MAX_CONTRIBUTED_DAYS
+        day_visits = bound_visits_per_week(self._day_visits, COUNT_MAX_CONTRIBUTED_DAYS)
+
+        day_counts = dict()
+
+        # Use the default epsilon value if it is not given as an argument
+        if not epsilon:
+            x = dp.CountInt(self._epsilon)
+        else:
+            x = dp.CountInt(epsilon)
+
```

Comment:
Here before calling `x.result(...)` we need to set l0 sensitivity on x. Otherwise the default of 1 will be used and that's not enough (the results will not be DP). This is the first case where the default sensitivity is incorrect, it needs to be set in most cases.